AOSCOSLauncher/AOSCOSLauncher.cpp

```diff
@@ -84,8 +101,43 @@ HRESULT SetDefaultUser(std::wstring_view userName)
     return hr;
 }
 
+static HRESULT GenerateLocaleCommand(LPWSTR command_buf)
+{
+    // Generate the command which sets container locale according to system locale.
+    // The buffer given to this function should be big as CMD_BUF_MAX_LENGTH.
+    // This might be an AOSC OS specific functionality.
+
+    int result = 0;
+    HRESULT hr = ERROR_SUCCESS;
+
+    // Get a buffer which stores locale information.
+    LPWSTR locale = (LPWSTR)malloc(LOCALE_NAME_MAX_LENGTH * 2);
+    if (locale == nullptr) {
+        // Failed to get a buffer, we have nothing to do but bailing out.
+        return CO_E_INIT_MEMORY_ALLOCATOR;
+    }
+    result = GetUserDefaultLocaleName(locale, LOCALE_NAME_MAX_LENGTH);
+    if (result <= 0) {
+        // GetUserDefaultLocaleName() returns the string length of the locale name.
+        // If it fails, fallback to en_US.
+        hr = StringCchCopyW(locale, LOCALE_NAME_MAX_LENGTH, L"en_US");

```

Comment:
```suggestion
        hr = StringCchCopyW(locale, LOCALE_NAME_MAX_LENGTH, L"en-US");

```

Windows Locale Names are based on [RFC 4646](https://datatracker.ietf.org/doc/rfc4646/), so subtags are distinguished and separated from one another by a hyphen.

Reply:
`locale` is later used by Linux, so it needs to be underscored before formatting to actual command. I will fix this later.